src/Service/RequestService.php

```diff
@@ -138,6 +137,20 @@ public function destroy()
         return true;
     }
 
+     /**
+     * @return boolean
+     */
+    public function destroyWithParams($params)
+    {
+        $obj = $this->resource;
+
+        $obj->getClient()
+            ->setResourceKey($obj::RESOURCE_NAME)
+            ->send($this->getResourcePath($obj->toArray($params)), 'DELETE');
+
+        return true;
```

Comment:
There is a possibility that this request won't succeed, I don't think return true *always* is a good idea